dali/pipeline/operators/reader/loader/loader.cc

```diff
@@ -30,4 +30,15 @@ DALI_SCHEMA(LoaderBase)
       R"code(Hint for how much memory to allocate per image.)code", 1048576);
 
 
+size_t start_index(const size_t shard_id,
+                   const size_t shard_num,
+                   const size_t size) {
+  const size_t remainder = size % shard_num;
+  if (shard_id < remainder) {
```

Comment:
Just ```size * shard_id / shard_num``` wont do?

Reply:
That is what was there before

Reply:
Do you mean use floating point division? I don't trust it to always give exactly the right integer part. If you mean integer, but first multiply size by shard_id, then I think that could work (although I think it produces the opposite of the proposed code - last shards get more elements, which I think is counterintuitively)

Reply:
I always assumed that we would "borrow" from the next epoch to complete the last batch of the previous epoch.  At larger batch sizes and a lot of epochs, the repetition starts to add up; for imagenet size with batch 16k and 90 epochs we end up doing ~1% more work than we're really getting "credit" for